pilot/pkg/bootstrap/server.go

```diff
@@ -1061,10 +1064,14 @@ func (s *Server) maybeCreateCA(caOpts *CAOptions) error {
 		if s.kubeClient != nil {
 			corev1 = s.kubeClient.CoreV1()
 		}
-		// May return nil, if the CA is missing required configs - This is not an error.
 		if s.CA, err = s.createIstioCA(corev1, caOpts); err != nil {
```

Comment:
Yes it is. As of now, createIstioCA still sets up DNS certs needed for control plane authentication. This will need to be refactored separately. Note that only one of them is started.

Reply:
Do you mean createIstioCA is still used to provision the DNS certs from K8s CA for Istiod and the webhooks? What's the difficulty of provisioning them from the external CA? I think it would be much better if we can use the external CA to issue all certs. Please add a TODO here if you are going to address that in followup PRs.

Reply:
Makes sense. However, there is a different configuration knob today ("global.pilotCertProvider == kubernates") that handles this. We may need to migrate the config. Not sure how to do this exactly. Will add a TODO in this PR and address this in a future PR.